### PR TITLE
UI Component: Floating Navigation Button

### PR DESCRIPTION
**Pull request by Meg979**

### CODE REVIEW

File: tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+  data class Navigation(
+    val leadingIcon: TarkaIcon? = null,
+    val text: String,
+  ) : TUIFloatingNavButtonContentType()
+
+  object Burger : TUIFloatingNavButtonContentType()
+}
+
+@Composable fun TUIFloatingNavButton(
```
Comment (askNilesh) [inline tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt:66]: Please add documentation

File: tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  modifier: Modifier = Modifier,
+  style: TUIFloatingNavButtonStyle = TUIFloatingNavButtonStyle.defaultStyle,
```
Comment (askNilesh) [inline tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt:68]: Please include Ui and screen shot test for this component

**Review by askNilesh — APPROVED** [review]

File: tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt (hunk 2 above)
Comment (Meg979) [reply tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt:75]: Will create a separate PR for that @askNilesh

File: tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        TUIFloatingNavBurgerButton()
+      }
+
+      is Navigation -> {
+        TUIFloatingNavTypeButton(contentType, style)
+      }
+
+      is List -> {
+        TUIFloatingNavListButton(contentType)
+      }
```
Comment (Younes-Charfaoui) [inline tarkaui/src/main/java/com/tarkalabs/uicomponents/components/tab/TUIFloatingNavButton.kt:101]:
```suggestion
      is Burger -> TUIFloatingNavBurgerButton()
      is Navigation -> TUIFloatingNavTypeButton(contentType, style)
      is List -> TUIFloatingNavListButton(contentType)
```

**Review by Younes-Charfaoui — APPROVED** [review]

**Pull request merged**